happiness-backend/api/dao/journal_dao.py

```diff
@@ -1,5 +1,11 @@
 from api.models import Journal
 
+def get_journal_by_id(entry_id):
+    """
+    Returns a Journal entry object by ID.
+    """
+    return Journal.query.filter(Journal.id == entry_id).first()
```

Comment:
Very small thing, but generally it is advised to use `.filter_by` for simpler queries like this one.  [Source](https://stackoverflow.com/a/2128558/16776130)